apps/one-drive/src/connectors/one-drive/share-point/item.ts

```diff
@@ -0,0 +1,38 @@
+import { env } from '@/common/env';
+import { MicrosoftError } from '@/common/error';
+import type { MicrosoftDriveItem } from './items';
+
+type GetItemParams = {
+  itemId: string;
+  token: string;
+  siteId: string;
+  driveId: string;
+};
+
+export const getItem = async ({
+  token,
+  siteId,
+  driveId,
+  itemId,
+}: GetItemParams): Promise<MicrosoftDriveItem | 'notFound'> => {
+  const url = new URL(`${env.MICROSOFT_API_URL}/sites/${siteId}/drives/${driveId}/items/${itemId}`);
+  url.searchParams.append('$select', 'id,folder,name,webUrl,createdBy,parentReference');
+
+  const response = await fetch(url, {
+    headers: {
+      Authorization: `Bearer ${token}`,
+    },
+  });
+
+  if (!response.ok) {
+    if (response.status === 404) {
+      return 'notFound';
+    }
+
+    throw new MicrosoftError('Could not retrieve item', { response });
+  }
```

Comment:
```suggestion
  if (response.status === 404) {
    return 'notFound';
  }
    
  if (!response.ok) {
    throw new MicrosoftError('Could not retrieve item', { response });
  }
```

Response.ok is false when status is not a 2XX only, so it's okay to check for 404 before checking if the response is ok.

Skippable: Regarding returning 'notFound', I personally found it cleaner to returns just `null`.

Reply:
good suggestion, updated